packages/legacy/core/App/components/misc/InactivityWrapper.tsx

```diff
@@ -0,0 +1,121 @@
+import { useAgent } from '@credo-ts/react-hooks'
+import React, { PropsWithChildren, useEffect, useRef } from 'react'
+import { AppState, PanResponder, View } from 'react-native'
+import { useAuth } from '../../contexts/auth'
+import { useStore } from '../../contexts/store'
+import { DispatchAction } from '../../contexts/reducers/store'
+import { TOKENS, useServices } from '../../container-api'
+
+export enum LockOutTime {
+  OneMinute = 1,
+  ThreeMinutes = 3,
+  FiveMinutes = 1,
+  Never = 0,
+}
+
+interface InactivityWrapperProps {
+  timeoutLength?: LockOutTime // number of minutes before timeoutAction is triggered, a value of 0 will never trigger the timeoutAction and an undefined value will default to 5 minutes
+}
+
+const InactivityWrapper: React.FC<PropsWithChildren<InactivityWrapperProps>> = ({ children, timeoutLength }) => {
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+  useStore()
+  const [_, dispatch] = useStore()
+  const { agent } = useAgent()
+  const { removeSavedWalletSecret } = useAuth()
+  const timeout_minutes = timeoutLength !== undefined ? timeoutLength : LockOutTime.OneMinute
+  const inactivityTimer = useRef<NodeJS.Timeout | null>(null)
+  const panResponder = React.useRef(
+    PanResponder.create({
+      onStartShouldSetPanResponderCapture: () => {
+        // some user interaction, reset timeout
+        resetInactivityTimeout(timeout_minutes)
+
+        // returns false so the PanResponder doesn't consume the touch event
+        return false
+      },
+    })
+  ).current
+
+  const resetInactivityTimeout = (minutes: number) => {
```

Comment:
You probably need to wrap all your fn inside a `useCallback`. Every time a component renders a function will be re-created which is a performance hit. Also, because a fn can be a dependency on a `useEffect` when it is re-created it will change causing a `useEffect` to trigger. So, in general, its good to just wrap 'em and let RN sort it out.

Reply:
functions are wrapped in callbacks